src/Zenith.sol

```diff
@@ -42,10 +50,10 @@ contract Zenith is HostPassage, AccessControlDefaultAdminRules {
     error BadSignature(address derivedSequencer);
 
     /// @notice Emitted when a new rollup block is successfully submitted.
+    /// @param location - the location where the block data was submitted.
     /// @param sequencer - the address of the sequencer that signed the block.
     /// @param header - the block header information for the block.
-    /// @param blobIndices - the indices of the 4844 blob hashes for the block data.
-    event BlockSubmitted(address indexed sequencer, BlockHeader indexed header, uint32[] blobIndices);
+    event BlockSubmitted(DataLocation indexed location, address indexed sequencer, BlockHeader indexed header);
```

Comment:
hmmm we can no longer extract the indices from the event, which means we need to the input to this contract, which may require a tx trace. can we set this up in a way that neither calldata nor blob require a trace? 

seems difficult 😮‍💨

Reply:
yeah i mean, this isn't a production solution but: https://github.com/init4tech/zenith/pull/7/commits/3d14eb1472a5c0ea858936a2472005a0868124a4

Reply:
we've aligned that we'll do this for now, and wean ourselves off it once calldata tracing is implemented